### DISCUSSION TITLE
Some Firebase features are not supported

### DISCUSSION THREAD

**Discussion by VietAnh1010** [joulev/ezkomment#117] [Questions]
In the declaration file, the signature of `update` is:

```ts
update(data: UpdateData, precondition?: Precondition): Promise<WriteResult>;
```

And the declaration of `Precodition` is:

```ts
export interface Precondition {
  readonly lastUpdateTime?: Timestamp;
  readonly exists?: boolean;
}
```

Adding the precondition, I get this error: `Error: Update() requires either a single JavaScript object or an alternating list of field/value pairs that can be followed by an optional precondition. Value for argument "preconditionOrValues" is not a valid precondition. "exists" is not an allowed precondition`.

It seems like this feature is not implemented at the moment, but it's already added into the declaration file.

I'm so tired with the documentations of Firebase Admin.

**Comment by joulev** [comment]
1. Please do not file stuff that aren't bugs/features as issues. Q/A are for Discussions.
2. Please only read the documentation and not the declaration file. We have no business touching those files unnecessarily.
3. Generally that means *you* are wrong and not Firebase. See e.g. [here](https://stackoverflow.com/q/62434737). Check your code. Firebase Admin SDK has [1.1m weekly downloads](https://www.npmjs.com/package/firebase-admin), they're not gonna leave such an obvious bug to the released code.
4. I can assure you that this app is not so complex that this precondition thing is needed. I reckon [`set()`](https://googleapis.dev/nodejs/firestore/latest/DocumentReference.html#set) and [`delete()`](https://googleapis.dev/nodejs/firestore/latest/DocumentReference.html#delete) are all what you'll need to interact with Firestore.
5. Please hurry up. The work you're doing should have been done before Milestone 1 (see the Milestone 1 description in the programme overview slide). We are 12 days away from Milestone 2 deadline, and we must implement the sites/pages/comment feature fully by then.

**Reply by VietAnh1010** [reply]
> Please hurry up. The work you're doing should have been done before Milestone 1 (see the Milestone 1 description in the programme overview slide). We are 12 days away from Milestone 2 deadline, and we must implement the sites/pages/comment feature fully by then.

Please start integrating. I will continue writing the API specification.

**Reply by joulev** [reply]
@VietAnh1010 No I can't. Your branch needs to be merged, then my branch updated with your code, only then can I continue. We cannot simply merge random branches together &ndash; I won't do that, that's not SWE.
